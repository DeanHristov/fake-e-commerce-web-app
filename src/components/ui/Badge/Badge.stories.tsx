import type { Meta, StoryObj } from '@storybook/react';

import Badge, { IBadgeProps } from './Badge';
import { CogIcon, ShoppingCartIcon } from '@heroicons/react/24/solid';

const meta: Meta<IBadgeProps> = {
  title: 'UI/Badge',
  component: Badge,
  parameters: {
    layout: 'centered',
  },
};

type Story = StoryObj<IBadgeProps>;

export const Primary: Story = {
  args: {
    counter: 1,
    children: <ShoppingCartIcon className="h-7 h-7 text-blue-500" />,
  },
};

export const WithDot: Story = {
  args: {
    dot: true,
    children: <CogIcon className="h-7 h-7 text-blue-500" />,
  },
};

export default meta;
