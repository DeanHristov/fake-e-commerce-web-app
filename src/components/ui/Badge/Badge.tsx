import { FC, ReactNode } from 'react';

export interface IBadgeProps {
  children: ReactNode;
  counter?: number;
  dot?: boolean;
  color?: string;
  onClick?: () => void;
}

const Badge: FC<IBadgeProps> = ({ children, onClick, counter, dot, color }) => {
  return (
    <span
      onClick={onClick}
      className="relative inline-block px-2 cursor-pointer"
    >
      {children}
      {counter && (
        <span
          className={`absolute top-0 right-2 inline-flex items-center justify-center px-2 py-1 text-xs font-bold leading-none text-red-100 transform translate-x-1/2 -translate-y-1/2 ${
            color ?? 'bg-red-600'
          } rounded-full`}
        >
          {counter}
        </span>
      )}
      {!counter && dot && (
        <span
          className={`absolute top-0 right-2 inline-block w-2 h-2 transform translate-x-1/2 -translate-y-1/2 ${
            color ?? 'bg-red-600'
          } rounded-full`}
        />
      )}
    </span>
  );
};

export default Badge;
