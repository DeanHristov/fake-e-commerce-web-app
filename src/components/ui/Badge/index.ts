import Badge, { IBadgeProps } from './Badge';

export type { IBadgeProps };

export default Badge;
