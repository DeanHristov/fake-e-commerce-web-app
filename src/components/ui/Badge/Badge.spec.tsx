import { render } from '@testing-library/react';
import Badge from './Badge';
import { ShoppingCartIcon } from '@heroicons/react/24/solid';

describe('UI/Component <Button {...} />', () => {
  it('Should render the badge with a counter', () => {
    const { container } = render(
      <Badge counter={6}>
        <ShoppingCartIcon className="h-7 h-7 text-gray-700 animate-shaking" />
      </Badge>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render the badge with a dot ', () => {
    const { container } = render(
      <Badge dot>
        <ShoppingCartIcon className="h-7 h-7 text-gray-700 animate-shaking" />
      </Badge>,
    );

    expect(container).toMatchSnapshot();
  });
});
