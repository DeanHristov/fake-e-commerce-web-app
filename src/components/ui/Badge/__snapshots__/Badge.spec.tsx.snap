// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI/Component <Button {...} /> Should render the badge with a counter 1`] = `
<div>
  <span
    class="relative inline-block px-2 cursor-pointer"
  >
    <svg
      aria-hidden="true"
      class="h-7 h-7 text-gray-700 animate-shaking"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.25 2.25a.75.75 0 000 1.5h1.386c.17 0 .318.114.362.278l2.558 9.592a3.752 3.752 0 00-2.806 3.63c0 .414.336.75.75.75h15.75a.75.75 0 000-1.5H5.378A2.25 2.25 0 017.5 15h11.218a.75.75 0 00.674-.421 60.358 60.358 0 002.96-7.228.75.75 0 00-.525-.965A60.864 60.864 0 005.68 4.509l-.232-.867A1.875 1.875 0 003.636 2.25H2.25zM3.75 20.25a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zM16.5 20.25a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0z"
      />
    </svg>
    <span
      class="absolute top-0 right-2 inline-flex items-center justify-center px-2 py-1 text-xs font-bold leading-none text-red-100 transform translate-x-1/2 -translate-y-1/2 bg-red-600 rounded-full"
    >
      6
    </span>
  </span>
</div>
`;

exports[`UI/Component <Button {...} /> Should render the badge with a dot  1`] = `
<div>
  <span
    class="relative inline-block px-2 cursor-pointer"
  >
    <svg
      aria-hidden="true"
      class="h-7 h-7 text-gray-700 animate-shaking"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2.25 2.25a.75.75 0 000 1.5h1.386c.17 0 .318.114.362.278l2.558 9.592a3.752 3.752 0 00-2.806 3.63c0 .414.336.75.75.75h15.75a.75.75 0 000-1.5H5.378A2.25 2.25 0 017.5 15h11.218a.75.75 0 00.674-.421 60.358 60.358 0 002.96-7.228.75.75 0 00-.525-.965A60.864 60.864 0 005.68 4.509l-.232-.867A1.875 1.875 0 003.636 2.25H2.25zM3.75 20.25a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0zM16.5 20.25a1.5 1.5 0 113 0 1.5 1.5 0 01-3 0z"
      />
    </svg>
    <span
      class="animate-ping absolute top-1.5 right-2 inline-block w-2 h-2 transform translate-x-1/2 -translate-y-1/2 bg-red-600 rounded-full"
    />
  </span>
</div>
`;
